include/bluetooth/mesh/sensor.h

```diff
@@ -64,7 +68,389 @@ enum bt_mesh_sensor_sampling {
 	BT_MESH_SENSOR_SAMPLING_COUNT,
 };
 
+/** Sensor sampling cadence */
+enum bt_mesh_sensor_cadence {
+	/** Normal sensor publish interval. */
+	BT_MESH_SENSOR_CADENCE_NORMAL,
+	/** Fast sensor publish interval. */
+	BT_MESH_SENSOR_CADENCE_FAST,
+};
+
+/** Unit for single sensor channel values. */
+struct bt_mesh_sensor_unit {
+	/** English name of the unit, e.g. "Decibel". */
+	const char *name;
+	/** Symbol of the unit, e.g. "dB". */
+	const char *symbol;
+};
+
+struct bt_mesh_sensor_format;
+
+/** Single sensor channel */
+struct bt_mesh_sensor_channel {
+	/** Format for this sensor channel. */
+	const struct bt_mesh_sensor_format *format;
+#ifdef CONFIG_BT_MESH_SENSOR_LABELS
+	/** Name of this sensor channel. */
+	const char *name;
+#endif
+};
+
+/** Flag indicating this sensor type has a series representation. */
+#define BT_MESH_SENSOR_TYPE_FLAG_SERIES BIT(0)
+
+/** Sensor type. Should only be instantiated in sensor_types.c.
+ *  See sensor_types.h for a list of all defined sensor types.
+ */
+struct bt_mesh_sensor_type {
+	/** Device Property ID. */
+	uint16_t id;
+	/** Flags, @see BT_MESH_SENSOR_TYPE_FLAG_SERIES */
+	uint8_t flags;
+	/** The number of channels supported by this type. */
+	uint8_t channel_count;
+	/** Array of channel descriptors.
+	 *
+	 *  All channels are mandatory and immutable.
+	 */
+	const struct bt_mesh_sensor_channel *channels;
+};
+
+struct bt_mesh_sensor;
+struct bt_mesh_sensor_srv;
+
+#if !defined(CONFIG_BT_MESH_SENSOR_USE_SENSOR_VALUE) || defined(__DOXYGEN__)
+/** Sensor value type representing the value and format of a single channel of
+ *  sensor data.
+ */
+struct bt_mesh_sensor_value {
+	/** The format the sensor value is encoded in. */
+	const struct bt_mesh_sensor_format *format;
+	/** Raw encoded sensor value, in little endian order. */
+	uint8_t raw[CONFIG_BT_MESH_SENSOR_CHANNEL_ENCODED_SIZE_MAX];
+};
+
 /** Sensor descriptor representing various static metadata for the sensor. */
+struct bt_mesh_sensor_descriptor {
+	/** Sensor measurement tolerance specification. */
+	struct {
+		/** Maximum positive measurement error (in steps of 1/4095 of
+		 *  100% error).
```

Comment:
Since for instance `bt_mesh_sensor_cli_series_entries_get` gets a significant amount of lines explaining how an inclusive range is used, I think that an example may be preferable here. 
The unclear part for me is "... error in steps of 1/4095 of 100% error".

Reply:
> Maximum postive measurement error, encoded according to MshMDL 4.1.1.2

This would be better, though it introduces an extra step in the process by having to look up the spec, which is cumbersome.

I remember being confused about the same thing when working on the sensor sample a year ago, and I think as stated that just adding the equation would make things more clear. 

A bit out of scope of the changes in this PR, but a solution could be to include more detail in an example in the sensor sample.